### ISSUE TITLE
FirstMethodArgumentLineBreak plays poorly with arrays and hashes

### ISSUE THREAD

**Issue by Epigene** [rubocop/rubocop#6493]
Maybe this is intended behavior, but it seems that `Layout/FirstMethodArgumentLineBreak` does nothing if the first argument is a Hash or Array literal.

Additionally, I'd like the closing parameter parenthesis to be aligned consistently, but `Layout/ClosingParenthesisIndentation` is no help in this (perhaps I need to use a different cop?).

--------

## Expected behavior

Given rubocop config:

```yaml
AllCops:
  DisabledByDefault: true 
  UseCache: false
  TargetRubyVersion: 2.3.4

Layout/AlignArray:
  Enabled: true

Layout/ClosingParenthesisIndentation:
  Enabled: true

Layout/FirstMethodArgumentLineBreak:
  Enabled: true

Layout/IndentationWidth:
  Enabled: true
  Width: 2
 
Layout/MultilineMethodCallBraceLayout:
  Enabled: true
  EnforcedStyle: new_line
```

And offending (contrived) code:

```rb
begin
  begin
    sheet.add_row([
      a[0],
      a[1],
    ])
  end
end

begin
  begin
    sheet.add_row({
      k1: :v1,
      k2: :v2
    })
  end
end
```

I'd expect the autofixed code to be:

```rb
begin
  begin
    sheet.add_row(
      [
        a[0],
        a[1],
      ]
    )
  end
end

begin
  begin
    sheet.add_row(
      {
        k1: :v1,
        k2: :v2
      }
    )
  end
end
```

## Actual behavior

```rb
begin
  begin
    sheet.add_row([
      a[0],
      a[1],
    ]
                 )
  end
end

begin
  begin
    sheet.add_row({
      k1: :v1,
      k2: :v2
    }
                 )
  end
end
```

## RuboCop version

```
$ rubocop -V
0.60.0 (using Parser 2.5.3.0, running on ruby 2.3.4 x86_64-darwin17)
```

**Comment by athornton2012** [comment]
@marcotc we came across this issue today as well! It is the exact same as @Epigene where hashes and arrays are not getting new lines in method calls. Do you think that a fix will be released soon?
Thanks!

**Comment by AlexWayfer** [comment]
Additional example:

```ruby
sheet.add_row(k1:
  v1)
```

**Comment by lamont-granquist** [comment]
Dunno if this would be considered a separate issue but heredocs don't seem to be considered multiline either:

```ruby
foo(<<~EOF
  here
EOF
   )
```

**Comment by allentsai** [comment]
I was writing a feature request, but found this issue already filed.  I agree with above.

After adding
```
Layout/FirstMethodArgumentLineBreak:
  Enabled: true

Layout/FirstHashElementLineBreak:
  Enabled: true

Layout/ClosingParenthesisIndentation:
  Enabled: true

Layout/MultilineHashBraceLayout:
  Enabled: true
  EnforcedStyle: new_line
```
We end up with an autocorrection as follows.
Before:
```
  def public_api_as_json(options = {})
    as_json({
      only: [:id, :nature, :deleted, :token],
      methods: [:definition, :description, :label, :size],
    }.merge(options))
  end
```
After:
```
  def public_api_as_json(options = {})
    as_json({
      only: [:id, :nature, :deleted, :token],
      methods: [:definition, :description, :label, :size],
    }.merge(options)
           )
  end
```
I would have liked to end up with
```
  def public_api_as_json(options = {})
    as_json({
      only: [:id, :nature, :deleted, :token],
      methods: [:definition, :description, :label, :size],
    }.merge(options))
  end
```
or
```
  def public_api_as_json(options = {})
    as_json(
      {
        only: [:id, :nature, :deleted, :token],
        methods: [:definition, :description, :label, :size],
      }.merge(options)
    )
  end
```

**Comment by allentsai** [comment]
The cop might be somewhat broken, rubocop does not complain about:
```
        with(:get_object,
             custom_audience.fb_id,
             fields: request_fields
            ).
```
I would expect it to return:
```
        with(
          :get_object,
          custom_audience.fb_id,
          fields: request_fields
        ).
```

**Comment by lamont-granquist** [comment]
still there AFAIK

**Comment by lamont-granquist** [comment]
confirmed bug still exists in rubocop 1.2.0

**Comment by AlexWayfer** [comment]
This bot…

**Comment by forthrin** [comment]
It's now autocorrected like so:

```ruby
sheet.add_row([
                a[0],
                a[1]
              ])

sheet.add_row({
                k1: :v1,
                k2: :v2
              })

```

Would personally prefer:

```ruby
sheet.add_row([
  a[0],
  a[1]
])

sheet.add_row({
  k1: :v1,
  k2: :v2
})
```

Might be configurable, but at least it's prettier than the OP. Issue is from 2018. Acceptable to close?
